### DISCUSSION TITLE
Prisma connector for PaaS like firestore?

### DISCUSSION THREAD

**Discussion by hyochan** [prisma/prisma#8263] [Q&A] [answered]
I am curious about building a Prisma connector for PasS like firebase's [firestore](https://firebase.google.com/docs/firestore/quickstart).

If this is possible, maybe we can use Prisma in a client like in [react](https://reactjs.org).

For example, 

```ts
import firebase from 'firebase/app';
import {PrismaClient} from 'prisma';

const prisma = new PrismaClient();

const createUser = async () => {
  await prisma.user.create({
    data: {
      name: 'user',
    }
  });
}

<button onClick={createUser} />
```
I am thinking of creating a user in [cloud firestore](https://firebase.google.com/docs/firestore/quickstart) using Prisma Client.
This isn't a feature request but I am just curious and want to ask the Prisma team how this thing can be achieved?

**Comment by janpio** [comment, marked as answer]
This depends on how you talk to Firestore. If there is an API that is SQL like, it would be "simple" to achieve something like this as we would have most of the plumbing in place - but still a multi month effort as the surface of a connector in Prisma is huge. It needs to support the full API we offer in Prisma Client. If it is a HTTP based API or even one that requires a specialized library or Client, the effort would be much higher as we do not have that yet and we would need to figure out how to do that.

There are quite a few connector feature requests already: https://github.com/prisma/prisma/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22topic%3A+connector%22 Firestore would be another one - feel free to open an issue if you can not find an existing one.

Generally we want to make this much easier over time, also to e.g. only support part of the API and so on, but we are not there yet and most probably will not be in the near future as we have other priorities.

**Reply by hyochan** [reply]
Thanks for the detailed explanation. I really appreciate it.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
